File: C/Singly Linked List.c
#include <stdio.h>
#include <stdlib.h>
struct Node {
	int data;
  	struct Node *next, *head;
};
void insertNode(struct Node *temp, int data) {
	struct Node *newNode = (struct Node*)malloc(sizeof(struct Node));
	newNode->data = data;
	if (temp == NULL) {
		struct Node *temp = newNode;
		newNode->next = NULL;
		return;
	}
	else {
  		while (temp->next != NULL) {
    		temp = temp->next;
  		}
  		temp->next = newNode;
  	}
}
void printList(struct Node *temp) {
	if (temp == NULL) {
		printf("List Empty");
		return;
	}
	else {
  		while (temp != NULL) {
    		printf("%d ", temp->data);
    		temp = temp->next;
  		}
  	}
}
void deleteNode(struct Node *temp, int nodeOffset) {
	struct Node *prev;
	int i;
	prev = temp;
	temp = temp->next;
	for (i=1; i<nodeOffset; i++) {
		prev = prev->next;
		temp = temp->next;
	}
	prev->next = temp->next;
}
int main() {
  	struct Node *obj = NULL;
  	int n, val, i;
  	printf("Enter the Number of Elements : ");
  	scanf("%d",&n);
  	for (i=0; i<n; i++){
    	printf("Enter the Element : ");
    	scanf("%d",&val);
    	insertNode(obj, val);
	}
    printf("Elements of the list are : ");
    printList(NULL);
    printf("\n");
    printf("Enter the index of element to delete : ");
    scanf("%d",&val);
    deleteNode(obj, val);
    printf("Elements of the list are : ");
    printList(obj);
    printf("\n");
    return 0;
    /*
  struct Node *one = NULL;
  struct Node *two = NULL;
  struct Node *three = NULL;

  // Allocate memory
  one = malloc(sizeof(struct Node));
  two = malloc(sizeof(struct Node));
  three = malloc(sizeof(struct Node));

  // Assign value values
  one->data = 1;
  two->data = 2;
  three->data = 3;

  // Connect nodes
  one->next = two;
  two->next = three;
  three->next = NULL;

  // printing node-value
  head = one;
  printList(head);*/
}

File: Java/Singly Linked List.java
import java.util.*;
public class LinkedList { 
    Node head;
    static class Node { 
        int data; 
        Node next; 
        Node(int d) { 
            data = d; 
            next = null; 
        } 
    }
    public static LinkedList insert(LinkedList list, int data) { 
        Node new_node = new Node(data); 
        if (list.head == null) { 
            list.head = new_node; 
        } 
        else {
            Node last = list.head; 
            while (last.next != null) { 
                last = last.next; 
            }
            last.next = new_node; 
        }
        return list; 
    }
    public static void delete(LinkedList list, int ind) {
        Node currNode = list.head;
        Node tempNode = list.head.next;
        for (int i=1; i<ind; i++) { 
            currNode = currNode.next;
            tempNode = tempNode.next;
        }
        currNode.next = tempNode.next;
    }
    public static void printList(LinkedList list) {
        Node currNode = list.head; 
        System.out.print("LinkedList: ");
        while (currNode != null) { 
            System.out.print(currNode.data + " "); 
            currNode = currNode.next; 
        }
        System.out.println(" ");
    }
    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
        int n,val;
        LinkedList obj = new LinkedList();
        System.out.print("Enter the Number of Elements : ");
        n = s.nextInt();
        for (int i=0; i<n; i++) {
            System.out.print("Enter the Element : ");
            val = s.nextInt();
            obj = insert(obj, val);
        }
        printList(obj);
        System.out.print("Enter the Index of Element to delete : ");
        val = s.nextInt();
        obj.delete(obj, val);
        printList(obj);
    } 
}